projects/01/Mux.hdl

```diff
@@ -14,5 +14,9 @@ CHIP Mux {
     OUT out;
 
     PARTS:
-    // Put your code here:
-}
\ No newline at end of file
+    Not(in=sel, out=notsel);
+    And(a=a, b=notsel, out=anotsel);
+
+    And(a=b, b=sel, out=bandsel);
+    Or(a=anotsel, b=bandsel, out=out);

```

Comment:
As we discussed on a call, this is the equivalent of `(a && !sel) || (b && sel)`. Maybe this table, with the rows in a different order from the one in the book, makes it clearer what’s going on:

a | b | sel | a && !sel | b && sel | (a && !sel) \|\| (b && sel)
--|---|-----|-----------|----------|----------------------------
0 | 0 | 0 | 0 (= a) | 0 | 0 (= a)
0 | 1 | 0 | 0 (= a) | 0 | 0 (= a)
1 | 0 | 0 | 1 (= a) | 0 | 1 (= a)
1 | 1 | 0 | 1 (= a) | 0 | 1 (= a)
0 | 0 | 1 | 0 | 0 (= b) | 0 (= b)
0 | 1 | 1 | 0 | 1 (= b) | 1 (= b)
1 | 0 | 1 | 0 | 0 (= b) | 0 (= b)
1 | 1 | 1 | 0 | 1 (= b) | 1 (= b)